clients/wallet/src/error.rs

```diff
@@ -8,16 +9,45 @@ pub enum Error {
 	InvalidSecretKey,
 	#[error("Error fetching horizon data: {0}")]
 	HttpFetchingError(#[from] FetchError),
-	#[error("Oracle returned error")]
-	OracleError,
 	#[error("Could not build transaction: {0}")]
 	BuildTransactionError(String),
-	#[error("Transaction submission failed: {0}")]
-	HorizonSubmissionError(String),
+	#[error("Transaction submission failed. Title: {title}, Status: {status}, Reason: {reason}, Envelope XDR: {envelope_xdr}")]
+	HorizonSubmissionError { title: String, status: u16, reason: String, envelope_xdr: String },
 	#[error("Could not parse string: {0}")]
 	Utf8Error(#[from] std::str::Utf8Error),
 	#[error("Could not decode XDR")]
 	DecodeError,
 	#[error("Could not sign envelope")]
 	SignEnvelopeError,
+
+	// Cache Errors
+	#[error("Cache Error: Could not extract sequence number from transaction: {0:?}.")]
+	UnknownSequenceNumber(TransactionEnvelope),
+
+	#[error("Cache Error: Sequence number {0} already in use.")]
+	SequenceNumberAlreadyUsed(SequenceNumber),
+
+	#[error("Cache Error: Failed to create directory for this path:{0}.")]
+	CreateDirectoryFailed(String),
+
+	#[error("Cache Error: Failed to create a file for {envelope:?} with path:{path}.")]
+	FileCreationFailed { path: String, envelope: TransactionEnvelope },
+
+	#[error("Cache Error: Failed to write a file for {envelope:?} with path:{path}.")]
+	WriteToFileFailed { path: String, envelope: TransactionEnvelope },
+
+	#[error("Cache Error: Failed to remove file: {0:?}")]
+	DeleteFileFailed(String),
+
+	#[error("Cache Error: File: {0:?} does not exist")]
+	FileDoesNotExist(String),
+
+	#[error("Cache Error: Cannot read file: {0:?}")]
+	ReadFileFailed(String),
+
+	#[error("Cache Error: Invalid file: {0:?}")]
+	InvalidFile(String),
+
+	#[error("Cache Error: Cannot decode xdr format for file: {0:?}")]
+	DecodeFileFailed(String),
```

Comment:
Nit: If all of these are 'Cache Errors' we could actually group them under a new sub enum called `CacheError` which makes it apparent that these are Cache errors without having to mention it in the error message. But this is up to you if you think it's better to change it that way, I don't have hard feelings about this @b-yap.

Reply:
I started with a separate `CacheErrors` at first, but then there's a `Result<_,`**_`Vec<Error>`_**`>` return type, which means iterating over the list of `CacheError`s and converting them to `Error`.

Instead, I have the `CacheError(` variant as you suggested originally, and I introduced a struct **`CacheError`** with its specific _ErrorKind_, the **_`CacheErrorKind`_**. 
I wanted to clean up its display, hence the `#[error(transparent)]`.

Reply:
The new implementation is great, good job 👍